lib/crypt.c

```diff
@@ -110,6 +110,18 @@ get_hashfn (const char *setting)
   return 0;
 }
 
+static int
+check_badsalt_chars (const char *setting)
+{
+  size_t i = 0;
+
+  for (i = 0; setting[i] != '\0'; i++)
+    if (setting[i] <= 0x20 || setting[i] >= 0x7f)
```

Comment:
Signed comparisons (except on the few architectures where `char` is unsigned by default). They will work OK in practice, but non-intuitively the 8-bit chars will be caught by the first check and not by the second. I don't have a preference on leaving this as-is vs. adding extra casts vs. adding a comment.

Reply:
Fixed in rebased commit.